packages/google_sign_in/test/google_sign_in_test.dart

```diff
@@ -2,103 +2,131 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
 import 'package:flutter/services.dart';
 import 'package:google_sign_in/google_sign_in.dart';
 import 'package:mockito/mockito.dart';
 import 'package:test/test.dart';
 
 void main() {
+  Map userData = {
+    "email": "john.doe@gmail.com",
+    "id": "8162538176523816253123",
+    "photoUrl": "https://lh5.googleusercontent.com/photo.jpg",
+    "displayName": "John Doe",
+  };
+  Map defaultResponses = {
+    'init': null,
+    'signInSilently': userData,
+    'signIn': userData,
+    'signOut': null,
+    'disconnect': null,
+  };
+
   group('$GoogleSignIn', () {
     GoogleSignIn googleSignIn;
-
     List<String> invokedMethods;
-    Map<String, dynamic> stubResponse;
+    Map responses;
 
     setUp(() {
+      responses = new Map.from(defaultResponses);
       MockPlatformChannel mockChannel = new MockPlatformChannel();
 
       invokedMethods = <String>[];
-      stubResponse = _failureResponse();
 
-      Function invocationRecorder = (Invocation invocation) async {
-        invokedMethods.add(invocation.positionalArguments[0]);
-        return stubResponse;
-      };
-      // Mock with and without optional 2nd argument.
-      when(mockChannel.invokeMethod(any)).thenAnswer(invocationRecorder);
-      when(mockChannel.invokeMethod(any, any)).thenAnswer(invocationRecorder);
+      dynamic answer(Invocation invocation) {
+        final method = invocation.positionalArguments[0];
+        invokedMethods.add(method);
+        final response = responses[method];
+        return new Future.value(response);
+      }
 
-      googleSignIn = new GoogleSignIn.private(channel: mockChannel);
-    });
+      when(mockChannel.invokeMethod(typed(any), typed(any))).thenAnswer(answer);
```

Comment:
Since these tests were written the `MethodChannel.setMockMethodCallHandler` method was added. I think we should use that so we can remove the dependency on package:mockito and rename `GoogleSignIn.private` to `GoogleSignIn._`. You can see this pattern used in the [shared_preferences](https://github.com/flutter/plugins/blob/master/packages/shared_preferences/test/shared_preferences_test.dart#L39) and [firebase_database](https://github.com/flutter/plugins/blob/master/packages/firebase_database/test/firebase_database_test.dart#L26) tests.